pkg/middleware/csrf_test.go

```diff
@@ -0,0 +1,114 @@
+package middleware
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestMiddlewareCSRF(t *testing.T) {
+	tests := []struct {
+		name        string
+		cookieName  string
+		method      string
+		origin      string
+		host        string
+		defaultPort string
+		code        int
+	}{
+		{
+			name:        "mismatched origin and host is forbidden",
+			cookieName:  "foo",
+			method:      "GET",
+			origin:      "http://notLocalhost",
+			host:        "localhost",
+			defaultPort: "80",
+			code:        http.StatusForbidden,
+		},
+		{
+			name:        "mismatched origin and host is NOT forbidden with a 'Safe Method'",
+			cookieName:  "foo",
+			method:      "TRACE",
+			origin:      "http://notLocalhost",
+			host:        "localhost",
+			defaultPort: "80",
+			code:        http.StatusOK,
+		},
+		{
+			name:        "mismatched origin and host is NOT forbidden without a cookie",
+			cookieName:  "",
+			method:      "GET",
+			origin:      "http://notLocalhost",
+			host:        "localhost",
+			defaultPort: "80",
+			code:        http.StatusOK,
```

Comment:
I'm probably stupid right now, but don't understand how this could be OK when origin and host is different?

Reply:
A missed the `name` spanning multiple rows :facepalm: